chia/types/blockchain_format/program.py

```diff
@@ -177,6 +177,12 @@ def match(o: SExp, expected: bytes) -> None:
     def as_int(self) -> int:
         return int_from_bytes(self.as_atom())
 
+    def as_atom(self) -> bytes:
+        ret: Optional[bytes] = self.atom
+        if ret is None:
+            raise ValueError("expected atom")
+        return ret
```

Comment:
@richardkiss I'm thinking of introducing the convention that `as_atom()` returns `bytes` and throws in case it's a pair. Just like `as_int()` throws right now it it's a pair.

There is a lot of demand for a function with this behavior. As I'm enabling type checking, I'd say most code that calls `node.atom` expects it to be `bytes` and virtually never checks for `None`. `as_atom()` isn't used very much, but I'm thinking of switching over call sites that expect the node to be an atom to use this function instead. It makes mypy happy and makes the exception thrown a bit more user friendly.